Modules/TPC/src/GenericHistogramCheck.cxx

```diff
@@ -0,0 +1,353 @@
+    }
+  } else {
+    mCheckStdDev = true;
+    ILOG(Warning, Support) << "No Check was given. The options are Range and StdDev. As a default StdDev was chosen." << ENDM;
+  }
+
+  if (const auto param = mCustomParameters.find("axis"); param != mCustomParameters.end()) {
+    const std::string axisString = param->second.c_str();
+    if (const auto param1 = axisString.find("X"); param1 != string::npos) {
+      mCheckXAxis = true;
+    }
+    if (const auto param1 = axisString.find("Y"); param1 != string::npos) {
+      mCheckYAxis = true;
+    }
+    if (!mCheckXAxis && !mCheckYAxis) {
+
+      mCheckXAxis = true;
+      ILOG(Warning, Support) << "The given value for axis was not readable. The options are X and Y. As a default X was chosen." << ENDM;
+    }
+  } else {
+    mCheckStdDev = true;
+    ILOG(Warning, Support) << "No axis was given. The options are X and Y. As a default X was chosen." << ENDM;
+  }
+
+  if (const auto param = mCustomParameters.find("ExpectedValueX"); param != mCustomParameters.end()) {
+    mExpectedValueX = std::atof(param->second.c_str());
+  } else {
+    mExpectedValueX = 0;
+    ILOG(Warning, Support) << "No ExpectedValueX was given. Please always give an expected value. As a default 0 was chosen. " << ENDM;
+  }
+  if (const auto param = mCustomParameters.find("ExpectedValueY"); param != mCustomParameters.end()) {
+    mExpectedValueY = std::atof(param->second.c_str());
+  } else {
+    mExpectedValueY = 0;
+    ILOG(Warning, Support) << "No ExpectedValueY was given. Please always give an expected value. As a default 0 was chosen. " << ENDM;
+  }
+
+  if (mCheckRange) {
+    if (const auto param = mCustomParameters.find("RangeX"); param != mCustomParameters.end()) {
+      mRangeX = std::atof(param->second.c_str());
+    } else {
+      mRangeX = 0.1 * mExpectedValueX;
+      ILOG(Warning, Support) << "No RangeX was given even though it was requested as a check. As a default " << 0.1 * mExpectedValueX << " was chosen. " << ENDM;
+    }
+    if (const auto param = mCustomParameters.find("RangeY"); param != mCustomParameters.end()) {
+      mRangeY = std::atof(param->second.c_str());
+    } else {
+      mRangeY = 0.1 * mExpectedValueY;
+      ILOG(Warning, Support) << "No RangeY was given even though it was requested as a check. As a default " << 0.1 * mExpectedValueY << " was chosen. " << ENDM;
+    }
+  }
+}
+Quality GenericHistogramCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  // ILOG(Warning, Support) << "Check started....?" << ENDM;
+  Quality result = Quality::Null;
+  for (auto const& moObj : *moMap) {
+    auto mo = moObj.second;
+    if (!mo) {
+      continue;
+      ILOG(Error, Support) << "No MO found" << ENDM;
+    }
+    const auto moName = mo->getName();
+
+    THn* hN = THn::CreateHn(moName.c_str(), moName.c_str(), (TH2F*)mo->getObject());
```

Comment:
Why is `hN` needed? It is not used except for the line below. I guess the dimension of a histogram can also be obtained directly from the histogram.
Also, the object does not seem to be deleted, so it's a memory leak...

Reply:
There is no way to even initiallize the object since we don't know if its an TH1 or TH2. The (TH2F*) also works for TH1s. when I try to initallize with auto it initiallizes it as a TObject not as a TH1/TH2.